0036-spsp-pull-payments/0036-spsp-pull-payments.md

```diff
@@ -0,0 +1,148 @@
+
+### Operation Overview
+
+The Payee's SPSP Client will set up a [STREAM](../0029-stream/0029-stream.md) connection to the Payer's SPSP Server using the Pull Payment Pointer, as described by the [Simple Payment Setup Protocol](../0009-simple-payment-setup-protocol/0009-simple-payment-setup-protocol.md). On connection, the SPSP Client tries to pull value from the SPSP server, staying within the terms of the Pull Payment Agreement.
+
+## Model of Operation
+
+### Negotiating the Pull Payment Agreement
+
+Prior to a Pull Payment, the Payer and the Payee have to negotiate a Pull Payment Agreement. This agreement SHOULD be comprised of the parameters defined in the `agreement`-object within the [Response Body](#Response-Body). The Payer's SPSP Server SHOULD store the negotiated Pull Payment Agreement parameters.
+
+The result of the negotiation is a Pull Payment Pointer representing the Pull Payment Agreement. The Pull Payment Pointer is required by the Payee's SPSP Client to connect to the Payer's SPSP server. 
+
+Implementations MAY try to parse the Pull Payment Pointer, however, it MUST support totally opaque Pull Payment Pointers. 
+
+### Conducting the Pull Payment
+
+The Payee's SPSP Client opens a [STREAM](../0029-stream/0029-stream.md) connection to the Payer's SPSP Server as described in the [Simple Payment Setup Protocol](../0009-simple-payment-setup-protocol/0009-simple-payment-setup-protocol.md). 
+
+Once this connection is established, the process continues as follows: 
+
+The SPSP Server begins sending ILP packets to complete the Pull Payment.
+  1. The SPSP Server will adjust their `sendMax` to reflect the amount they're willing to send.
+      * `sendMax` SHOULD be derived from the Pull Payment Agreement parameters, i.e., `sendMax` SHOULD be equal to `pull.balance.available`, converted to the SPSP Server's operating asset, taking exchange rate fluctuations into account.
+  2. The SPSP Client will adjust their `receiveMax` to reflect the amount they're willing to receive.
+      * `receiveMax` MAY be `Infinity`.
+  3. The SPSP Client's and Server's [STREAM Modules](../0009-simple-payment-setup-protocol/0009-simple-payment-setup-protocol.md#Definitions) will move as much value as possible while staying inside these bounds.
+  4. If the SPSP Server reaches their `sendMax`, they end the stream and the connection. If the SPSP Client reaches their `receiveMax`, they will end the stream and the connection.
+
+The STREAM parameters - `sendMax` and `receiveMax` - are defined in [STREAM's frame encoding](../0029-stream/0029-stream.md#53-frames).
+
+Note that if there are multiple open STREAM connections to the `destination_account` the Pull Payment Pointer is resolving to, the SPSP Server MUST only set one `sendMax` at a time.
```

Comment:
```suggestion
Note that if there are multiple open streams (via one or more STREAM connections) to the `destination_account` the Pull Payment Pointer is resolving to, the SPSP Server MUST only set one `sendMax` at a time.
```
Thinking about this some more, you could technically set lower `sendMax`'s across multiple simultaneous streams and make sure the sum of them doesn't exceed the amount you're willing to send.
But I think this is fine as is since we're already saying this above:
>The SPSP Server will adjust their `sendMax` to reflect the amount they're willing to send.